### PR TITLE
Feature/return early stop

### PR DESCRIPTION
**Pull request by vindolineKis**
Add return value for `EarlyStop`.

### CODE REVIEW

File: QCircLearning/utils.py [hunk 1]
```diff
@@ -49,9 +49,9 @@ def __init__(self, patience=5, min_delta=0.0, verbose=False):
         self.verbose = verbose
         self.best_loss = None
         self.counter = 0
-        self.early_stop = False
+        # self.early_stop = False
 
-    def __call__(self, current_loss):
+    def __call__(self, current_loss, stop=False):
```
Comment (cfenglv) [inline QCircLearning/utils.py]: why there will be a stop=False ?

File: QCircLearning/utils.py [hunk 2]
```diff
@@ -62,4 +62,5 @@ def __call__(self, current_loss):
             if self.verbose:
                 print(f"EarlyStopping counter: {self.counter}/{self.patience}")
             if self.counter >= self.patience:
-                self.early_stop = True
+                stop = True
```
Comment (cfenglv) [inline QCircLearning/utils.py]: Directly return False here.

File: QCircLearning/utils.py [hunk 3]
```diff
@@ -62,4 +62,5 @@ def __call__(self, current_loss):
...
+        return stop
```
Comment (cfenglv) [inline QCircLearning/utils.py]: Directly return True here.

**Review by cfenglv — CHANGES_REQUESTED** [review]

File: QCircLearning/utils.py (hunk 2 above)
Comment (cfenglv) [inline QCircLearning/utils.py]: Directly return True here.

File: QCircLearning/utils.py (hunk 3 above)
Comment (cfenglv) [inline QCircLearning/utils.py]: Directly return False here.

File: QCircLearning/utils.py (hunk 1 above)
Comment (vindolineKis) [reply QCircLearning/utils.py]: I want make sure it is not True at beginning

Comment (cfenglv) [reply QCircLearning/utils.py]: There is no such scenario that needs to be checked True or false at the beginning

File: QCircLearning/utils.py (hunk 3 above)
Comment (vindolineKis) [reply QCircLearning/utils.py]: This indeed seems more reasonable and simplifies the logic. I made the change accordingly. Thannnnnk U！

Comment (cfenglv) [reply QCircLearning/utils.py]: Possible misplace of earlystop initialization in trainermodel: Line 91-93. Earlystop should be for every training session at each iteration, not all of them. 
Suggestion: move into iteration for-loop

File: QCircLearning/trainerModel.py [hunk 4]
```diff
@@ -113,10 +110,14 @@ def NN_opt(func, x0, callback=None, **kwargs):
                     f"Run ID: {kwargs['run_id']}, Iteration {iteration + 1}/{max_iter}"
                 )
                 sys.stdout.flush()
+            # start_time = time.time()
```
Comment (cfenglv) [inline QCircLearning/trainerModel.py]: Remove the time record part as they are irrelevant to Feature/early-stop

Comment (vindolineKis) [reply QCircLearning/trainerModel.py]: okay

**Review by cfenglv — APPROVED** [review]

**Pull request merged**